### PR TITLE
Migrate docs gh pages

### LINKED ISSUE #127: [docs] Migrate docusaurus deployment from surge to gh-pages
### Is your feature request related to a problem? Please describe.
It will be better to have the docs on the github-pages.

### Describe the solution you'd like
Deploy the docs on gh-pages.
Take reference from this: anitab-org/mentorship-backend@4cc78ae

### PR DESCRIPTION
**Pull request by wizardlekic**
### Description

Migrating docusaurus deployment from surge to GitHub pages.
Referenced from: https://github.com/anitab-org/mentorship-backend/commit/4cc78ae
Also referenced: https://docusaurus.io/docs/deployment#triggering-deployment-with-github-actions

The check part of the workflow works, but I'm having trouble with deployment, I probably have to fiddle around some more with setting up the secret key on my own repository properly.

For deployment the secret `GH_PAGES_DEPLOY` needs to be added. I have not been successful with trying this out on my own repository, I'll keep investigating some more tomorrow.

**Update:** I've successfully managed to deploy the docs to https://wizardlemon.github.io/anitab-forms-backend/
The problem was that the docusaurus config file needed to be updated as well.

Fixes #127

### Type of Change:

- Quality Assurance
- Documentation

**Code/Quality Assurance Only**
- This change requires a documentation update (software upgrade on readme file)

### How Has This Been Tested?

I've tested the workflow by creating a PR to my own develop branch, the checks run successfuly. I then merged the file to test the deployment, but I have issues with deployment that I think are related to secret keys.

**Update:** I've successfully managed to deploy the docs to https://wizardlemon.github.io/anitab-forms-backend/
The problem was that the docusaurus config file needed to be updated as well.

### Checklist:

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [ ] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

**Code/Quality Assurance Only**
- [x] My changes generate no new warnings

### CODE REVIEW

**Comment by wizardlekic** [comment]
I've managed to successfully deploy the documentation to https://wizardlemon.github.io/anitab-forms-backend/ using the workflow actions after updating the docusaurus config file.

**Review by isabelcosta — DISMISSED** [review]
Great work! Thank you for providing a lot of context in the PR description ⭐ That is really great!

Thank you for your contribution @WizardLemon !

File: .github/workflows/docs.yml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+          git config --global user.name "gh-actions"
+          cd osp-backend-docusaurus/
+          if [ -e yarn.lock ]; then
+          yarn install --frozen-lockfile
+          elif [ -e package-lock.json ]; then
+          npm ci
+          else
+          npm i
+          fi
+          npm run deploy
```
Comment (codesankalp) [inline .github/workflows/docs.yml]: blank line at the end of file

File: .github/workflows/docs.yml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+      - uses: webfactory/ssh-agent@v0.5.3
+        with:
+          ssh-private-key: ${{ secrets.GH_PAGES_DEPLOY }}
+      - name: Release to GitHub Pages
+        env:
+          USE_SSH: true
+          GIT_USER: git
+        run: |
+          git config --global user.email "actions@github.com"
+          git config --global user.name "gh-actions"
```
Comment (codesankalp) [inline .github/workflows/docs.yml]:
```suggestion
          git config --global user.email "opensource@anitab.org"
          git config --global user.name "anitab-org"
```

File: osp-backend-docusaurus/docusaurus.config.js [hunk 3]
```diff
@@ -1,13 +1,13 @@
 module.exports = {
   title: 'Open Source Programs Backend Docs',
```
Comment (codesankalp) [inline osp-backend-docusaurus/docusaurus.config.js]:
```suggestion
  title: 'Anitab forms backend Docs',
```

File: osp-backend-docusaurus/docusaurus.config.js [hunk 4]
```diff
@@ -1,13 +1,13 @@
...
   tagline: 'Documentation for Open Source Programs Backend',
```
Comment (codesankalp) [inline osp-backend-docusaurus/docusaurus.config.js]:
```suggestion
  tagline: 'Documentation for Anitab Forms Backend',
```

File: osp-backend-docusaurus/docusaurus.config.js [hunk 5]
```diff
@@ -1,13 +1,13 @@
...
-  url: 'https://osp-backend-docs.surge.sh/',
-  baseUrl: '/',
+  url: 'https://anitab-org.github.io/anitab-forms-backend/',
```
Comment (codesankalp) [inline osp-backend-docusaurus/docusaurus.config.js:4]:
```suggestion
  url: 'https://anitab-org.github.io',
```

**Review by codesankalp — CHANGES_REQUESTED** [review]
@WizardLemon See my comments :point_down:

**Comment by wizardlekic** [comment]
@codesankalp thanks for taking a thorough look and suggesting improvements, I've implemented the requested changes

File: .github/workflows/docs.yml [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+    if: github.event_name != 'push'
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          node-version: '12.x'
+      - name: Test Build
+        run: |
+          cd osp-backend-docusaurus/
```
Comment (codesankalp) [inline .github/workflows/docs.yml]: One more change I forgot to mention in the previous review, as we are not using osp-backend anymore, you can rename this directory in root directory as well as here with anitab-forms-backend.

**Comment by wizardlekic** [comment]
It seems I've had some conflicts after the newest changes, I've rebased my branch on top of those changes and updated accordingly.

**Review by isabelcosta — APPROVED** [review]
Looks good to me :) Thank you for working on this @WizardLemon 🙌🏾

**Pull request closed without merging**